capstone-app/src/app/modules/maps/map/map.component.ts

```diff
@@ -89,12 +89,61 @@ export class MapComponent implements OnInit{
     this.activeMarkerOutput.emit(this.activeMark);
   }
 
+  getCurrentLocation () {
+    let filterResult: string;
+    let prevResult: string = 'blank';
+    navigator.geolocation.getCurrentPosition(position => {
+      this.position = {
+        lat: position.coords.latitude,
+        lng: position.coords.longitude,
+      };
+      this.location = new google.maps.LatLng(this.position);
+      this.geocoder.geocode({'location': this.position}, (results, status) => {
+        if (status === 'OK') {
+          for (let result of results) {
+            filterResult = result.formatted_address;
+            if (filterResult.match(/[1-1000]/)) {
+              continue;
+            }
+            if (filterResult.includes('County')){
+              continue;
+            }
+            if (!filterResult.includes(',')) {
+              continue;
+            }
+            if (filterResult.split(',').length > 3) {
+              continue;
+            }
+            if (filterResult.split(',').length > prevResult.split(',').length) {
```

Comment:
What is the purpose of this one?

Reply:
the purpose of that one is to get formatted that is Miami, Fl, USA to be set for our cityLocation because if this isn't done then city location will set as Florida, USA so this pretty much get the option where the address has city, state, Country format instead of state, country format